include/deal.II/matrix_free/matrix_free.h

```diff
@@ -2290,51 +2305,50 @@ MatrixFree<dim, Number>::get_face_category(const unsigned int macro_face) const
 
 
 
-template <int dim, typename Number>
+template <int dim, typename Number, typename VectorizedArrayType>
 inline bool
-MatrixFree<dim, Number>::indices_initialized() const
+MatrixFree<dim, Number, VectorizedArrayType>::indices_initialized() const
 {
   return indices_are_initialized;
 }
 
 
 
-template <int dim, typename Number>
+template <int dim, typename Number, typename VectorizedArrayType>
 inline bool
-MatrixFree<dim, Number>::mapping_initialized() const
+MatrixFree<dim, Number, VectorizedArrayType>::mapping_initialized() const
 {
   return mapping_is_initialized;
 }
 
 
 
-template <int dim, typename Number>
-AlignedVector<VectorizedArray<Number>> *
-MatrixFree<dim, Number>::acquire_scratch_data() const
+template <int dim, typename Number, typename VectorizedArrayType>
+AlignedVector<VectorizedArrayType> *
+MatrixFree<dim, Number, VectorizedArrayType>::acquire_scratch_data() const
 {
   using list_type =
-    std::list<std::pair<bool, AlignedVector<VectorizedArray<Number>>>>;
+    std::list<std::pair<bool, AlignedVector<VectorizedArrayType>>>;
   list_type &data = scratch_pad.get();
   for (typename list_type::iterator it = data.begin(); it != data.end(); ++it)
     if (it->first == false)
       {
         it->first = true;
         return &it->second;
       }
-  data.push_front(
-    std::make_pair(true, AlignedVector<VectorizedArray<Number>>()));
+  data.push_front(std::make_pair(true, AlignedVector<VectorizedArrayType>()));
```

Comment:
Unrelated to this change, but since we are here please change to
```suggestion
  data.emplace_front(true, AlignedVector<VectorizedArrayType>());
```